GenderPayGap.WebUI/Classes/Extensions/HtmlHelpers.cs

```diff
@@ -62,9 +63,11 @@ public static async Task<IHtmlContent> PartialModelAsync<T>(this IHtmlHelper htm
         public static string WithQuery(this IUrlHelper helper, string actionName, object routeValues)
         {
             var newRoute = new NameValueCollection();
-            foreach (string key in helper.ActionContext.HttpContext.Request.Query.Keys)
+
+            if (helper.ActionContext.HttpContext.Request.QueryString.HasValue)
             {
-                newRoute[key] = helper.ActionContext.HttpContext.Request.Query[key];
+                var existingRoute = HttpUtility.ParseQueryString(helper.ActionContext.HttpContext.Request.QueryString.Value);
+                newRoute.Add(existingRoute);

```

Comment:
What does it do if you have a weird query string like
```
?param1=value1&param1=value2,value3
```
?

Reply:
"value2,value3" is considered a single value - as expected.

The issue was with the query string builder.
We had "p=v1&p=v2" and when building `newRoute` we ended up with `newRoute[p]` = `v1,v2` (In theory it should be iterable (we iterate through that list of values - line 79) but for some reason it was seen as a single value)

Basically now we do the same thing (build a name value collection) using a different method.